src/vdaf/mastic.rs

```diff
@@ -379,32 +391,485 @@ where
+                Ok(MasticPrepareShare::FirstRound(
+                    Seed::<SEED_SIZE>::decode(bytes)?,
+                    SzkQueryShare::<F, SEED_SIZE>::decode_with_param(
+                        &(*requires_joint_rand, *verifier_len),
+                        bytes,
+                    )?,
+                ))
+            }
+            MasticPrepareState::LaterRound(_, _) => {
+                Ok(MasticPrepareShare::LaterRound(Seed::<SEED_SIZE>::decode(
+                    bytes,
+                )?))
+            }
+        }
+    }
+}
+
+/// Mastic prepare message
+///
+/// Result of preprocessing the broadcast messages of all parties during the
+/// preparation phase.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MasticPrepareMessage<F: FieldElement, const SEED_SIZE: usize> {
+    ///If Szk is being run, all SzkQueryShares have been summed
+    /// to produce a verifier to be input to decide()
+    FirstRound(SzkVerifier<F, SEED_SIZE>),
+    /// If Szk is not being run, no further computation is necessary as the VIDPF proofs have already
+    /// been checked.
+    LaterRound,
+}
+
+impl<F: FieldElement, const SEED_SIZE: usize> Encode for MasticPrepareMessage<F, SEED_SIZE> {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        match self {
+            MasticPrepareMessage::FirstRound(verifier) => verifier.encode(bytes),
+            MasticPrepareMessage::LaterRound => Ok(()),
+        }
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        match self {
+            MasticPrepareMessage::FirstRound(verifier) => verifier.encoded_len(),
+            MasticPrepareMessage::LaterRound => Some(0),
+        }
+    }
+}
+
+impl<F: FieldElement, const SEED_SIZE: usize> ParameterizedDecode<MasticPrepareState<F, SEED_SIZE>>
+    for MasticPrepareMessage<F, SEED_SIZE>
+{
+    fn decode_with_param(
+        prep_state: &MasticPrepareState<F, SEED_SIZE>,
+        bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        match prep_state {
+            MasticPrepareState::FirstRound(_, _, verifier_len, requires_joint_rand) => {
+                Ok(MasticPrepareMessage::FirstRound(
+                    SzkVerifier::decode_with_param(&(*requires_joint_rand, *verifier_len), bytes)?,
+                ))
+            }
+            MasticPrepareState::LaterRound(_, _) => Ok(MasticPrepareMessage::LaterRound),
+        }
+    }
+}
+
+impl<T, P, const SEED_SIZE: usize> Aggregator<SEED_SIZE, NONCE_SIZE> for Mastic<T, P, SEED_SIZE>
+where
+    T: Type,
+    P: Xof<SEED_SIZE>,
+{
+    type PrepareState = MasticPrepareState<T::Field, SEED_SIZE>;
+
+    /// The type of messages sent by each aggregator at each round of the Prepare Process.
+    ///
+    /// Decoding takes a [`Self::PrepareState`] as a parameter; this [`Self::PrepareState`] may be
+    /// associated with any aggregator involved in the execution of the VDAF.
```

Comment:
No need to include this comment since this is a method required by a trait we're implementing. Here and below.
```suggestion
```